### ISSUE TITLE
[BUG]  sizing_mode="stretch_both" fails since 3.0.0

### ISSUE THREAD

**Issue by attilabibok** [bokeh/bokeh#12627]
### Software versions

Python version      :  3.9.13 (main, Aug 18 2022, 19:15:15) 
IPython version     :  (not installed)
Tornado version     :  6.2
Bokeh version       :  3.0.2
BokehJS static path :  /home/abibok/kisters/repos/service/venv/lib/python3.9/site-packages/bokeh/server/static
node.js version     :  v19.0.1
ERROR: [Errno 2] No such file or directory: 'npm'

### Browser name and version

Google Chrome Version 107.0.5304.110 (Official Build) (64-bit)

### Expected behavior

Vertical sizing matches the size of the frame, if sizing_mode="stretch_both" is set in a figure

### Observed behavior

Vertical sizing fails, if sizing_mode="stretch_both" is set in a figure. Regardless of the setup, it fails even with the "graph layout example". Just adding that single line, the vertical axis collapses (see the screenshots).

### Example code

```Python
import math


from bokeh.models import Ellipse, GraphRenderer, StaticLayoutProvider
from bokeh.palettes import Spectral8
from bokeh.plotting import figure, show

N = 8
node_indices = list(range(N))
plot = figure(
    title="Graph Layout Demonstration",
    sizing_mode="stretch_both",
    x_range=(-1.1, 1.1),
    y_range=(-1.1, 1.1),
    tools="",
    toolbar_location=None,
)

graph = GraphRenderer()

graph.node_renderer.data_source.add(node_indices, "index")
graph.node_renderer.data_source.add(Spectral8, "color")
graph.node_renderer.glyph = Ellipse(height=0.1, width=0.2, fill_color="color")

graph.edge_renderer.data_source.data = dict(start=[0] * N, end=node_indices)

# create a static layout
circ = [i * 2 * math.pi / 8 for i in node_indices]
x = [math.cos(i) for i in circ]
y = [math.sin(i) for i in circ]
graph_layout = dict(zip(node_indices, zip(x, y)))
graph.layout_provider = StaticLayoutProvider(graph_layout=graph_layout)

# draw quadratic bezier paths
def bezier(start, end, control, steps):
    return [
        (1 - s) ** 2 * start + 2 * (1 - s) * s * control + s**2 * end for s in steps
    ]


xs, ys = [], []
sx, sy = graph_layout[0]
steps = [i / 100.0 for i in range(100)]
for node_index in node_indices:
    ex, ey = graph_layout[node_index]
    xs.append(bezier(sx, ex, 0, steps))
    ys.append(bezier(sy, ey, 0, steps))
graph.edge_renderer.data_source.data["xs"] = xs
graph.edge_renderer.data_source.data["ys"] = ys

plot.renderers.append(graph)

show(plot)
```

### Screenshots

Expected:
![image](https://user-images.githubusercontent.com/22617734/202858802-62a6e0d2-eb87-4bfd-8ba0-cc0ec382cd3f.png)

Observed:

![image](https://user-images.githubusercontent.com/22617734/202858707-d6e8f05e-f61a-448a-a3e3-c7d3398b3367.png)

**Comment by mosc9575** [comment]
Looks like a duplicate of #12614.

**Comment by mattpap** [comment]
> Looks like a duplicate of https://github.com/bokeh/bokeh/issues/12614.

Definitely in height, but not necessarily in width.

**Comment by FelixFest** [comment]
Is this fixed now? I updated to 3.1.0 and I still get the same result.

**Comment by leidenheit** [comment]
I can confirm that this does affect BokehJS v3.2.0 too.
